onboarding_auth/onboardingAuthDomain/src/main/java/com/alexafit/onboardingauthdomain/useCase/SetAuthorizationTokenUseCase.kt

```diff
@@ -0,0 +1,11 @@
+package com.alexafit.onboardingauthdomain.useCase
+
+import com.alexafit.onboardingauthdata.repository.OnboardingAuthRepository
+
+class SetAuthorizationTokenUseCase(
+    private val repository: OnboardingAuthRepository
+) {
+    suspend operator fun invoke(authToken: String?) {
+        repository.setDataStoreAuthKey(authToken)
+    }
+}
```

Comment:
Use cases like these will lead to a lot of overarchitecting in my experience. Do you really want to add such an empty use case for every little thing you could do on the data layer? Like setting the key, getting the key, setting the user's full name, getting the name, logging in, etc...
All these don't do anything, but cause a lot of extra effort when things change in your codebase. It's totally fine to directly use `OnboardingAuthRepository` in your ViewModels, since that's a domain dependency and not bound to data. Then it's probably not clean architecture as in the books, but if 80% of your use cases don't do anything else than calling another class's function, that feels everything but right to me.
I personally isolate real business logic in such use case like classes (like calculating the user's initials for example), but for such simple calls I just use the dependency right away in the ViewModel.